### PR TITLE
Add user research recruitment banner

### PR DESCRIPTION
**Pull request by chrisroos**
Trello: https://trello.com/c/yGMU3sR2

This adds a "sticky" banner on the Signon home page to recruit user research participants for GOV.UK Publishing. The "Sign up to take part in research" button opens a Google Form (managed by the User Research folk) in a new browser tab. If the user clicks on this button, we record that in the Signon database and we never show them the banner again. If the user clicks on the "Hide this" link, then we hide the banner for the duration of the current browser session, but it will be displayed to the user in their next browser session.

@chrisroos Wondered whether there's a better way to style the "Hide this" button as a link? But I haven't found any information about doing that, so I've left it as it was.

I've the following changes to @chrisroos' original work

* Open Google form in a new browser tab
* Use GOV.UK Design System responsive spacing helpers
* Add padding to button group to balance the bottom of the banner with the top
* Improve the vertical alignment of the "Sign up..." & "Hide this" buttons
* Change button text -> "Find out more" based on content design feedback

As per [these recommendations](https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices#browsers-to-test-in-from-june-2022), I've tested against the following devices/browsers (using BrowserStack):

* [x] Windows 11
  * [x] Edge 115
  * [x] Chrome 115
  * [x] Firefox 116
* [x] MacOS 13.4.1
  * [x] Firefox 115
  * [x] Chrome 115
  * [x] Safari 16.5.2
* [x] iPhone SE 2022 v15.4
  * [x] Safari
  * [x] Chrome
* iPhone 14 iOS 16
  * [x] Safari
* [x] Galaxy S23 Android 13
  * [x] Chrome
  * [x] Samsung Internet

### Desktop
Screen Shot 2023-08-08 at 12 23 34

### iPad
Screen Shot 2023-08-08 at 12 24 10

### iPhone
Screen Shot 2023-08-08 at 12 24 46

### CODE REVIEW

File: test/integration/user_research_recruitment_banner_test.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    user = create(:user, name: "user-name", email: "user@example.com")
+    visit new_user_session_path
+    signin_with(user)
+
+    click_on "Change your email or password"
+
+    assert_not has_content?(user_research_recruitment_banner_title)
+  end
+
+  should "hide the banner until the next session"
```
Comment (mike29736) [inline test/integration/user_research_recruitment_banner_test.rb]: I'm re-learning MiniTest right now. I was expecting this blockless `should` to count as a skipped test in the test runner's report (a technique I'm familiar with from outside-in testing with other tools), but it doesn't. But I'm assuming that this is a fairly common approach?

File: app/assets/stylesheets/_user_research_recruitment_banner.scss [hunk 2]
```diff
@@ -16,3 +16,10 @@
 .user-research-recruitment-banner__intro {
   color: govuk-colour("white");
 }
+
+.user-research-recruitment-banner__hide-this {
```
Comment (mike29736) [inline app/assets/stylesheets/_user_research_recruitment_banner.scss:24]: I'm not sure what to think about this! Like, is there a reason we can't find anything in the design system to help us out here?

File: app/controllers/user_research_recruitment_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class UserResearchRecruitmentController < ApplicationController
```
Comment (mike29736) [inline app/controllers/user_research_recruitment_controller.rb:1]: This commit's the first one where I get the feeling that I'm following the development process and not seeing the feature fall into place step-by-step.

**Review by mike29736 — APPROVED** [review]
The banner looks weird on my machine
Screenshot_2023-08-08_16-57-41
but I assume I haven't, like, setup something clientside properly in my dev environment? I trust the combination of your two's machines and BrowserStack.

I left a comment about the commit order. Re-ordering those early commits would take some work, but squashing the last few commits wouldn't. It's very personal taste stuff, though, so it's fine either way with me!

File: app/controllers/user_research_recruitment_controller.rb (hunk 3 above)
Comment (floehopper) [reply app/controllers/user_research_recruitment_controller.rb:1]: I've resolved this by curating the commits in #2301.

**Comment by floehopper** [comment]
@mike29736 

Thanks for reviewing this.

> The banner looks weird on my machine

As discussed earlier, you need to run `rails assets:precompile` and restart your app if you're running the app in the govuk-docker environment.

Closing in favour of the reworked version in #2301

**Pull request closed without merging**